### ISSUE TITLE
Clangd Trace

### ISSUE THREAD

**Issue by zrno** [clangd/vscode-clangd#190]
There is a setting in vs code for clangd: Trace. _Names a file that clangd should log a performance trace to, in chrome trace-viewer JSON format._

I couldn't find any explanation about this. How do I trigger the generation of this file?

**Comment by kadircet** [comment]
You just set it to a file path you'd want clangd to output tracing info (note that clangd won't create a directory, but it will create the file, or overwrite it).

If that fails that's definitely a bug and I can't test it at the moment, but you can also try setting `CLANGD_TRACE` env variable while starting vscode. e.g: `CLANGD_TRACE=/tmp/clangd_trace.json code /path/to/my/repo`

**Comment by zrno** [comment]
when should the file be generated? Is it generated when I build a CMake target? Is it generated when I build using any compiler?

**Comment by kadircet** [comment]
hmm, i hope you are aware that we are talking about clang**d** and not clang here.

it will be created once you start your editing session (i.e. vscode starts clangd)

**Comment by zrno** [comment]
yep, I am aware it is clangd. I am trying to find out if I am doing something wrong because the file doesn't get generated.

**Comment by HighCommander4** [comment]
I did try this locally on Linux, and the setting works fine for me.

If you're not able to get it to work, could you share:
 * What OS you're on
 * What value you're using for the setting
 * The clangd server logs (available in the Output window)

**Comment by zrno** [comment]
![image](https://user-images.githubusercontent.com/13214299/121266720-222d0400-c90f-11eb-9505-2a9f3ea76088.png)
Note: Creating a file in that location manually from a terminal works.

Clangd server logs when I change the value of the settings and press enter:
![image](https://user-images.githubusercontent.com/13214299/121266800-41c42c80-c90f-11eb-8385-1b101c400768.png)


Extension version: 0.1.11
VS Code version: Code 1.56.2 (054a9295330880ed74ceaedda236253b4f39a335, 2021-05-12T16:45:26.313Z)
OS version: Linux x64 5.12.9-arch1-1

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 3000)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|31.06GB (5.30GB free)|
|Process Argv|--no-sandbox --unity-launch --crash-reporter-id 87fed376-385e-4b70-a6e7-f62fe0f386e3|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|wayland|
</details>

**Comment by sam-mccall** [comment]
The performance trace file option is unrelated to LSP's $/setTrace method.

> Clangd server logs when I change the value of the settings and press enter:

Did you restart clangd after changing the setting? (e.g. with the "restart clangd" command, or just by restarting vscode)

**Comment by zrno** [comment]
I didn't restart at all. Just entered a new value and pressed enter. Should I restart?
